src/uucp_break.ml

```diff
@@ -35,6 +35,38 @@ let sentence u = Array.unsafe_get Low.sentence_of_int (Low.sentence u)
 
 let east_asian_width u = Uucp_rmap.get Uucp_break_data.east_asian_width_map u
 
+let err_ctrl_char e =
+  invalid_arg (Printf.sprintf "0x%02X is a control character" e)
+
+let err_not_uchar e =
+  invalid_arg (Printf.sprintf "U+%04X is not a unicode scalar value." e)
+
+let terminal_width_hint =
+  let gc = Uucp_gc.general_category in
+  function
+  (* C0 (without 0x00) + DELETE + C1 is an error. *)
+  |u when 0 < u && u <= 0x1f || 0x7f <= u && u <= 0x9f -> err_ctrl_char u
```

Comment:
On one hand, it seems to be somehow.. customary to treat **C0** and **C1** not as characters, but as special codes. It makes things simpler to describe, I guess.

On the other hand, U+85, U+0A and U+0D are most likely to really move the cursor, making their width undefined, but U+2028 and U+2029 won't:

![paras-st](https://cloud.githubusercontent.com/assets/325535/10951233/4cb82424-8334-11e5-8a6d-67dc8d3f0a5c.png)
![paras-urxvt](https://cloud.githubusercontent.com/assets/325535/10951234/4cb85138-8334-11e5-8ebf-946925df86c6.png)
![paras-xterm](https://cloud.githubusercontent.com/assets/325535/10951235/4cb887ca-8334-11e5-91ee-3c4ffe199151.png)
![paras-gnome-terminal](https://cloud.githubusercontent.com/assets/325535/10951232/4cb50938-8334-11e5-904d-d37cb6974799.png)

One could argue whether it is more likely for U+2028 and U+2029 to be 0 or 1-wide.